src/test/scala/me/jeffmay/neo4j/client/ws/WSNeo4jClientBasicSpecs.scala

```diff
@@ -0,0 +1,94 @@
+package me.jeffmay.neo4j.client.ws
+
+import me.jeffmay.neo4j.client._
+import me.jeffmay.util.UniquePerClassNamespace
+import me.jeffmay.util.akka.TestGlobalAkka
+import org.scalatest._
+import play.api.libs.json.Json
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class WSNeo4jClientBasicSpecs extends fixture.AsyncWordSpec
+  with Safety
+  with UniquePerClassNamespace
+  with CleanupNamespace {
+
+  override implicit def executionContext: ExecutionContext = ExecutionContext.global
+
+  class FixtureParam extends UniqueNamespace {
+    val client: Neo4jClient = new WSNeo4jClient(
+      TestNeo4jWSClient.TestWS,
+      Neo4jClientConfig.fromConfig(TestGlobal.config),
+      TestGlobalAkka.scheduler,
+      executionContext
+    )
+  }
+
+  override def withAsyncFixture(test: OneArgAsyncTest): Future[Outcome] = {
+    val fixture = new FixtureParam
+    val futureOutcome = test(fixture)
+    val x = futureOutcome.andThen {
+      case _ => cleanupNamespace(fixture.namespace, fixture.client)
+    }
+    x
+  }
+
+  "WSNeo4jClient" when {
+
+    "providing authorization" should {
+
+      "not require a password change when auto-initialize is turned on" in { fixture =>
+        import fixture._
+        for {
+          passwordChangeRequired <- client.passwordChangeRequired()
+        } yield {
+          assertResult(false, "password change should not be required after initializing") {
+            passwordChangeRequired
+          }
+        }
+      }
+    }
+
+    "calling openAndCloseTxn" should {
+
+      "start with an empty unique namespace" in { fixture =>
+        import fixture._
+        val getAllNodes = CypherStatement(
+          "MATCH (n { ns: {props}.ns }) RETURN n",
+          CypherParameters(Json.obj("ns" -> namespace))
+        )
+        for {
+          rsp <- client.openAndCommitTxn(Seq(getAllNodes))
+        } yield {
+          assertResult(true, "should not encounter errors") {
+            rsp.errors.isEmpty
+          }
+          assertResult(Seq(), "database should be empty") {
+            // TODO: Results by index is ok, but not as ideal as a map
+            rsp.results.head.data
+          }
```

Comment:
@egprentice async tests! Super simple. If the future completes with a test failed exception then it prints the first error. You just return a `Future[Outcome]` and there is conversions from `Unit` => `Outcome`, `Boolean` => `Outcome`, etc. If you extend `Safety` then it won't convert `Any` => `Outcome`, so you can be assured that your test is actually testing something (although unit is not necessarily the safest assumption, because I could have a test that just calls `println` and not assert anything, but meh... I like the simplicity of the API).